### PR TITLE
webui: devices view

### PR DESCRIPTION
**Pull request by crcro**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

basic view:
screen shot 2016-10-08 at 10 13 55 pm

detail view:
screen shot 2016-10-11 at 8 28 05 pm

### CODE REVIEW

**Comment by laf** [comment]
That's awesome work @crcro :)

I would say though I'm not a fan of the taller status boxes and prefer the old way ( @librenms/reviewers )?

image

**Comment by crcro** [comment]
main reason was to keep things within same aspect ratio ...

**Comment by paulgear** [comment]
I would rather keep the status boxes centred the way they are now.  As it stands, I would probably never use the one line version.  I think it leaves out too much info.  If we're trying to make a useful one-line version, I would prefer to do it by reducing the space between the elements and separating out the columns (i.e. separate columns for ports & sensors, separate column for OS & version, etc.).

**Comment by laf** [comment]
Changing the number of columns between the basic / detail view would add unnecessary complexity to the code so sticking  with the same # of columns across both would be my preferred way.

I can only imaging the basic view had more of a use when it wasn't a paginated page as it reduces the queries run. However I do think it looks good, it's a lot cleaner and simple. Gets my vote.

**Comment by crcro** [comment]
what @paulgear said has made me thinking about adding more one-line info in basic view.
@laf ... changing the columns between view is not such a big complexity, but no pagination would add a possible pitfall for long lists of devices?

**Comment by laf** [comment]
It'll increase the complexity of the codebase for no real benefit imho. 

Don't remove the pagination though

File: html/pages/devices.inc.php [hunk 1]
```diff
@@ -10,96 +10,83 @@
...
-        echo(" selected");
-    }
     $display_type = is_mib_graph($type, $avail_type) ? $avail_type : nicecase($avail_type);
-    echo(">$display_type</option>");
+    if ('graph_' . $avail_type == $vars['format']) {
+        $is_selected = 'selected';
+    } else {
+        $is_selected = '';
+    }
+    $graphs_types .= "<option value=" . generate_url($vars, array('format' => 'graph_' . $avail_type)) . " $is_selected >$display_type</option>";
```
Comment (Jellyfrog) [inline html/pages/devices.inc.php]: missing quotes around `value`

Comment (crcro) [inline html/pages/devices.inc.php]: fixed in latest push

File: html/pages/devices.inc.php [hunk 2]
```diff
@@ -210,264 +197,254 @@
...
+    </div>
+    ';
 
-<?php
+    ?>
+
+    <script>
+        var grid = $("#devices").bootgrid({
+            ajax: true,
+            rowCount: [5, 10, 50, 100, 250, -1],
```
Comment (laf) [inline html/pages/devices.inc.php]: This defaults to 5 now which is too small, 50 made more sense.

File: html/pages/devices.inc.php [hunk 3]
```diff
@@ -210,264 +197,254 @@
...
+            columnSelection: false,
+            formatters: {
+                "status": function (column, row) {
```
Comment (laf) [inline html/pages/devices.inc.php]: This is still showing a bigger box with small text - I think you just need to revert the status box back imho.

**Comment by laf** [comment]
This imho is still too low a number, it's now 10 devices per page and should be 50.

Also the text in the box is still too small.

**Comment by laf** [comment]
All good from my perspective.

@librenms/reviewers ?

**Comment by murrant** [comment]
Only thing I could think that could be improved is allow for user selectable columns.  (check the all ports page for an example)

**Comment by crcro** [comment]
@murrant besides vendor, metrics and actions columns all of them are sortable, do you need anything more, or maybe i didn't understood your request.

**Comment by laf** [comment]
See my commit I just pushed.

**Comment by murrant** [comment]
@crco Go to Ports, click the button on the far right, there are columns that aren't shown by default...  But are able to be shown.

**Pull request merged**